dll/shellext/sendmail/CDeskLinkDropHandler.cpp

```diff
@@ -86,14 +86,48 @@ CDeskLinkDropHandler::Drop(IDataObject *pDataObject, DWORD dwKeyState,
         StringCbCopyW(szDest, sizeof(szDest), szDir);
         if (SHGetPathFromIDListW(pidl, szSrc))
         {
-            CStringW strTitle;
-            strTitle.Format(IDS_SHORTCUT, PathFindFileNameW(szSrc));
-
-            PathAppendW(szDest, strTitle);
-            PathRemoveExtensionW(szDest);
-            StringCbCatW(szDest, sizeof(szDest), L".lnk");
-
-            hr = CreateShellLink(szDest, szSrc, NULL, NULL, NULL, NULL, -1, NULL);
+            LPCWSTR pszSourceExt;
+            BOOL bIsLink;
+
+            pszSourceExt = PathFindExtensionW(szSrc);
+            bIsLink = (_wcsicmp(pszSourceExt, L".lnk") == 0);
+
+            if (bIsLink)
+            {
+                PathAppendW(szDest, PathFindFileNameW(szSrc));
+            }
+            else
+            {
+                CStringW strTitle;
+                strTitle.Format(IDS_SHORTCUT, PathFindFileNameW(szSrc));
+
+                PathAppendW(szDest, strTitle);
+                PathRemoveExtensionW(szDest);
+                StringCbCatW(szDest, sizeof(szDest), L".lnk");
+            }
+
+            if (PathFileExistsW(szDest))
+            {
+                CStringW strName(PathFindFileNameW(szDest));
+                PathYetAnotherMakeUniqueName(szDest, szDir, NULL, strName);
+            }
+
+            if (bIsLink)
+            {
+                hr = (CopyFileW(szSrc, szDest, TRUE) ? S_OK : E_FAIL);
```

Comment:
What happens here if the link is already on the desktop, and you copy the link to the desktop again?

Reply:
![Capture2](https://user-images.githubusercontent.com/86486978/204835135-26fd22b5-339e-4b0f-848b-138fc98f219b.PNG)